FaunaDB.Client/Encoding/Encoder.cs

```diff
@@ -0,0 +1,244 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using FaunaDB.Query;
+using FaunaDB.Types;
+using FaunaDB.Attributes;
+using System.Linq.Expressions;
+
+namespace FaunaDB.Encoding
+{
+    public static class Encoder
+    {
+        public static Expr Encode(object obj) =>
```

Comment:
I think this should work at the `Value` level. I don't see a reason why automatic serialization should return `Expr`. Do you have any thoughts on that?